tests/utilities/test_apply_func_torchtext.py

```diff
@@ -0,0 +1,170 @@
+import torch
+import torchtext
+
+import pytorch_lightning as pl
+
+
+def get_torchtext_data_iterator(include_lengths=False):
+    text_field = torchtext.data.Field(sequential=True, pad_first=False,  # nosec
+                                      init_token="<s>", eos_token="</s>", include_lengths=include_lengths)  # nosec
+
+    example1 = torchtext.data.example.Example.fromdict({"text": "a b c a c"}, {"text": ("text", text_field)})
+    example2 = torchtext.data.example.Example.fromdict({"text": "b c a a"}, {"text": ("text", text_field)})
+    example3 = torchtext.data.example.Example.fromdict({"text": "c b a"}, {"text": ("text", text_field)})
+
+    dataset = torchtext.data.Dataset([example1, example2, example3], {"text": text_field})
+    text_field.build_vocab(dataset)
+
+    iterator = torchtext.data.Iterator(dataset, batch_size=3,
+                                       sort_key=None, device=None, batch_size_fn=None,
+                                       train=True, repeat=False, shuffle=None, sort=None, sort_within_batch=None)
+    return iterator, text_field
+
+
+def test_move_data_to_device_torchtext_include_length_true():
+    """Test if batches created by torchtext with include_lengths=True raise an exception."""
+
+    class DebugModel(pl.LightningModule):
```

Comment:
which part of the test module is specific to this task? mind rather write specific methods to Template and assigned them here...

Reply:
I am not sure if I fully understand what you are suggesting. 

The tests are derived to test the two modes that torchtext uses when generating batches with and without length information included. Do you suggest to refactor the model and iterator into a separate file from the model fit, or do you suggest adding more tests that are specific to changes in apply_func.py?

Reply:
The test is too complex with this whole model definition. To test that the batch object is moved to the device properly, you can do without defining a whole new model.

Reply:
The motivation was to test that torchtext `Field` with `include_lengths=True` and PL work together. I can try to create a more targeted test for the `function move_data_to_device` just moving data to device. However, I might need some advice on how to todo that.

I believe that the more complex test using the full model has value once if torchtext changes the interface. Where would you put such a test?

Reply:
maybe we need a template model for a torchtext task like the RNN we already have. @Borda what do you th ink?

Reply:
yes, sorry that it was not so clear... I see two ways
 - if the change is minimal, just add a method to existing Template
 - add a next template which can be easily adjusted for other use-cases

Reply:
@Borda @awaelchli : When you refer to "Template" I am not sure what that means, and where they are located. 

I added two more specific tests, and think the PR is getting close to be ready. Maybe we can discuss via Slack if `test_apply_func_torchtext.py` should be split in two separate test, or the specific tests should be moved to an already existing test file. One motivation for a new file was to not introduce a dependency of torchtext into the existing tests. You advice is very much appreciated.

Reply:
mind check https://github.com/PyTorchLightning/pytorch-lightning/blob/3f2c1022ab0e7027822a2acd66debc8afa90d5d6/tests/base/model_template.py#L19

Reply:
Thanks @Borda and @awaelchli. As discussed (Slack) I removed models from test but kept the specific test for data transfer. Will create a separate PR for testing an RNN model using `torchtext` to generate batches with `include_lengrths=True`.